processor/src/advice/dsa.rs

```diff
@@ -0,0 +1,68 @@
+use super::{ExecutionError, Felt, StarkField, Vec, Word};
+use vm_core::{
+    crypto::dsa::rpo_falcon512::{KeyPair, Polynomial},
+    utils::Deserializable,
+};
+
+/// Gets as input a vector containing an expanded public key and its associated secret key, and a
+/// word representing a message and outputs a vector of values to be pushed onto the advice stack.
+/// The values are the ones required for a Falcon signature verification inside the VM and they are:
+///
+/// 1. The nonce represented as 8 field elements.
+/// 2. The expanded public key represented as the coefficients of a polynomial of degree < 512.
+/// 3. The signature represented as the coefficients of a polynomial of degree < 512.
+/// 4. The product of the above two polynomials in the ring of polynomials with coefficients
+/// in the Miden field.
+///
+/// # Errors
+/// Will return an error if either:
+/// - The advice map does not contain an entry with key PK.
+/// - The advice map entry under key PK is not a vector of the expected length.
+///
+/// The function only generates non-deterministic input that is required for the Falcon verification
+/// procedure inside the VM and as such does interact with the VM only through the advice provider.
```

Comment:
nit: I think this can probably be deleted.

Reply:
Yep!